addons/survey/controllers/main.py

```diff
@@ -416,10 +416,9 @@ def survey_section_get_background(self, survey_token, section_id):
         return self._get_background_image(section._name, section.id)
 
     def _get_background_image(self, model_name, res_id):
-        status, headers, image_base64 = request.env['ir.http'].sudo().binary_content(
-            model=model_name, id=res_id, field='background_image',
-            default_mimetype='image/png')
-        return request.env['ir.http']._content_image_get_response(status, headers, image_base64)
+        record = request.env[model_name].sudo().browse(res_id)
+        stream = request.env['ir.binary']._get_image_stream_from(record, 'background_image')
+        return stream.get_response()
```

Comment:
It looks like the way `_get_background_image` was defined matched the interface for `binary_content`.

That is not necessary anymore, so the record should be passed in directly (with no sudoing inside here). Possibly the entire thing should be inlined, as at this point it becomes a relatively simple call but YMMV.